evap/staff/views.py

```diff
@@ -801,12 +814,58 @@ def questionnaire_copy(request, questionnaire_id):
         else:
             return render(request, "staff_questionnaire_form.html", dict(form=form, formset=formset))
     else:
-        questionnaire = get_object_or_404(Questionnaire, id=questionnaire_id)
+        form, formset = get_identical_form_and_formset(questionnaire_id)
+        return render(request, "staff_questionnaire_form.html", dict(form=form, formset=formset))
+
+
+def find_questionnaire_name(old_questionnaire):
+    timestamp = datetime.date.today()
+    new_name_de = '{} (until {})'.format(old_questionnaire.name_de, str(timestamp))
+    new_name_en = '{} (until {})'.format(old_questionnaire.name_en, str(timestamp))
+
+    if Questionnaire.objects.filter(Q(name_de=new_name_de) | Q(name_en=new_name_en)):
+        return None, None
+    return new_name_de, new_name_en
+
+
+@staff_required
+def questionnaire_new_version(request, questionnaire_id):
+    if request.method == "POST":
+        questionnaire = Questionnaire()
         InlineQuestionFormset = inlineformset_factory(Questionnaire, Question, formset=AtLeastOneFormSet, form=QuestionForm, extra=1, exclude=('questionnaire',))
 
-        form = QuestionnaireForm(instance=questionnaire)
-        formset = InlineQuestionFormset(instance=questionnaire, queryset=questionnaire.question_set.all())
+        form = QuestionnaireForm(request.POST, instance=questionnaire)
+        formset = InlineQuestionFormset(request.POST.copy(), instance=questionnaire, save_as_new=True)
```

Comment:
Did you execute the tests? I just died after removing and got the following error:

ERROR: test_failing_questionnaire_copy (evap.evaluation.tests.test_coverage.URLTests)
Traceback (most recent call last):
  File "/vagrant/evap/evaluation/tests/test_coverage.py", line 137, in test_failing_questionnaire_copy
    response = self.get_submit_assert_200("/staff/questionnaire/2/copy", "evap")
  File "/vagrant/evap/evaluation/tests/test_utils.py", line 47, in get_submit_assert_200
    response = response.forms[2].submit("")
  File "/usr/local/lib/python3.4/dist-packages/webtest/forms.py", line 607, in submit
    params=fields, **args)
  File "/usr/local/lib/python3.4/dist-packages/webtest/response.py", line 294, in goto
    return method(href, **args)
  File "/usr/local/lib/python3.4/dist-packages/django_webtest/__init__.py", line 125, in post
    upload_files, expect_errors, content_type, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/webtest/app.py", line 371, in post
    content_type=content_type)
  File "/usr/local/lib/python3.4/dist-packages/webtest/app.py", line 736, in _gen_request
    expect_errors=expect_errors)
  File "/usr/local/lib/python3.4/dist-packages/django_webtest/__init__.py", line 74, in do_request
    response = super(DjangoTestApp, self).do_request(req, status, expect_errors)
  File "/usr/local/lib/python3.4/dist-packages/webtest/app.py", line 606, in do_request
    res = req.get_response(app, catch_exc_info=True)
  File "/usr/local/lib/python3.4/dist-packages/webob/request.py", line 1295, in send
    application, catch_exc_info=True)
  File "/usr/local/lib/python3.4/dist-packages/webob/request.py", line 1263, in call_application
    app_iter = application(self.environ, start_response)
  File "/usr/local/lib/python3.4/dist-packages/webtest/lint.py", line 198, in lint_app
    iterator = application(environ, start_response_wrapper)
  File "/usr/local/lib/python3.4/dist-packages/django/contrib/staticfiles/handlers.py", line 63, in __call__
    return self.application(environ, start_response)
  File "/usr/local/lib/python3.4/dist-packages/django/core/handlers/wsgi.py", line 177, in __call__
    response = self.get_response(request)
  File "/usr/local/lib/python3.4/dist-packages/django/core/handlers/base.py", line 230, in get_response
    response = self.handle_uncaught_exception(request, resolver, sys.exc_info())
  File "/usr/local/lib/python3.4/dist-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.4/dist-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/vagrant/evap/evaluation/auth.py", line 104, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/vagrant/evap/staff/views.py", line 816, in questionnaire_copy
    return render(request, "staff_questionnaire_form.html", dict(form=form, formset=formset))
  File "/usr/local/lib/python3.4/dist-packages/django/shortcuts.py", line 67, in render
    template_name, context, request=request, using=using)
  File "/usr/local/lib/python3.4/dist-packages/django/template/loader.py", line 97, in render_to_string
    return template.render(context, request)
  File "/usr/local/lib/python3.4/dist-packages/django/template/backends/django.py", line 95, in render
    return self.template.render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 206, in render
    return self._render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/test/utils.py", line 92, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/loader_tags.py", line 173, in render
    return compiled_parent._render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/test/utils.py", line 92, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/loader_tags.py", line 173, in render
    return compiled_parent._render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/test/utils.py", line 92, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/loader_tags.py", line 173, in render
    return compiled_parent._render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/test/utils.py", line 92, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/loader_tags.py", line 69, in render
    result = block.nodelist.render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/loader_tags.py", line 69, in render
    result = block.nodelist.render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 992, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 959, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 1043, in render
    output = self.filter_expression.resolve(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 709, in resolve
    obj = self.var.resolve(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 850, in resolve
    value = self._resolve_lookup(context)
  File "/usr/local/lib/python3.4/dist-packages/django/template/base.py", line 913, in _resolve_lookup
    current = current()
  File "/usr/local/lib/python3.4/dist-packages/django/forms/formsets.py", line 281, in non_form_errors
    self.full_clean()
  File "/usr/local/lib/python3.4/dist-packages/django/forms/formsets.py", line 338, in full_clean
    form = self.forms[i]
  File "/usr/local/lib/python3.4/dist-packages/django/utils/functional.py", line 33, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/usr/local/lib/python3.4/dist-packages/django/forms/formsets.py", line 144, in forms
    for i in range(self.total_form_count())]
  File "/usr/local/lib/python3.4/dist-packages/django/forms/formsets.py", line 144, in <listcomp>
    for i in range(self.total_form_count())]
  File "/usr/local/lib/python3.4/dist-packages/django/forms/models.py", line 885, in _construct_form
    form.data[form.add_prefix(self._pk_field.name)] = None
  File "/usr/local/lib/python3.4/dist-packages/django/http/request.py", line 407, in __setitem__
    self._assert_mutable()
  File "/usr/local/lib/python3.4/dist-packages/django/http/request.py", line 404, in _assert_mutable
    raise AttributeError("This QueryDict instance is immutable")
AttributeError: This QueryDict instance is immutable